### ISSUE TITLE
Error when working with groups

### ISSUE THREAD

**Issue by Kathrin84** [jhoopes/moodle-mod_activequiz#22]
Hi,

when I choose the setup that students can work in groups, I make a new quiz, add questions to it, start a new session and then this error occurs:

Debug info: ERROR: invalid input syntax for integer: ""
SELECT * FROM mdl_activequiz_attempts WHERE forgroupid = $1 AND status = $2 AND sessionid = $3
[array (
0 => '',
1 => 10,
2 => '1',
)]
Error code: dmlreadexception
Stack trace:
line 443 of /lib/dml/moodle_database.php: dml_read_exception thrown
line 244 of /lib/dml/pgsql_native_moodle_database.php: call to moodle_database->query_end()
line 764 of /lib/dml/pgsql_native_moodle_database.php: call to pgsql_native_moodle_database->query_end()
line 606 of /mod/activequiz/classes/activequiz_session.php: call to pgsql_native_moodle_database->get_records_sql()
line 153 of /mod/activequiz/classes/controllers/view.php: call to mod_activequiz\activequiz_session->can_take_quiz_for_group()
line 38 of /mod/activequiz/view.php: call to mod_activequiz\controllers\view->handle_request()

This error makes it impossible to use your plugin with groups... :(

Kind regards,
Kathrin

**Comment by jhoopes** [comment]
Hi @Kathrin84 ,

Interesting bug this one.  I can't reproduce it, but it's likely because I'm using mysql, and judging from your error message you're using postgres.  

I guess mysql doesn't care if I pass an empty string for an integer column :)  

I believe that I've fixed this for you.  You'll have to re-download the master branch from github (or apply the changes from my commit, https://github.com/jhoopes/moodle-mod_activequiz/commit/31bbe7f64f0d61bd60e01f800209f5c4d0e93567) , and then clear the Moodle cache on your own, as I added a language string but didn't upgrade the plugin, and so the string cache won't be cleared.  

The commit is a simple change, but added a check for group id so that students can't bypass the take quiz for group check.  Although there should also probably be a check to make sure the group id is a valid group id in the first place.

Let me know, and I'll keep it there, as well as back fix it for the 2.7 and 2.8 branches.
